stats/statsvariables.cpp

```diff
@@ -1270,6 +1302,40 @@ struct WeightVariable : public StatsVariableTemplate<StatsVariable::Type::Numeri
 	}
 };
 
+// ============ Dive number in 10, 50, 100, 200 bins ============
+
+struct DiveNrBinner : public IntRangeBinner<DiveNrBinner, IntBin> {
+	DiveNrBinner(int bin_size) : IntRangeBinner(bin_size)
+	{
+	}
+	QString name() const override {
+		return StatsTranslations::tr("in %L2 steps").arg(bin_size);
+	}
+	int to_bin_value(const dive *d) const {
+		return d->number / bin_size;
+	}
+};
+
+static DiveNrBinner dive_nr_binner_10(10);
+static DiveNrBinner dive_nr_binner_50(50);
+static DiveNrBinner dive_nr_binner_100(100);
+static DiveNrBinner dive_nr_binner_200(200);
```

Comment:
interesting set of bins - because of our privacy policy I don't know what the actual distribution of total number of dives is, but when people give me access to fix things, we seem to have a lot of people in the <200 dives group - so having 5, 10, 20, 50 as bin sizes would seem useful. Having 100 & 200 for the people with tons and tons of dives isn't wrong, but I think they are more of an outlier (yes, one person that asked me for help had dive numbers past 6000, but they, too, had only about 800 dives in Subsurface)

Reply:
The list of binners is now different for logs with < 1000 and > 1000 dives. For the latter small bins result in horrible results.

Ultimately, the memory management should be changed to allow for dynamically generated bins. Not a big deal, but also not a high priority.

The code is also improved by now starting the bins at 10*n+1, since 0 means no number. Moreover, dives with number 0 are ignored.